### PR TITLE
HDDS-10691. CRYPTO_COMPLIANCE tag for cryptography parameters

### PR DESCRIPTION
**Pull request by Galsza**
## What changes were proposed in this pull request?
For crypto compliance mode the cryptography related configuration options need to be whitelisted in some way. After some deliberate consideration, the CRYPTO_COMPLIANCE tag is used and in [HDDS-10604](https://issues.apache.org/jira/browse/HDDS-10604) the whitelisting will be added for these options.
In this PR the CRYPTO_COMPLIANCE tag is added to all already existing configuration options.

`hdds.key.algo` and `hdds.security.provider` are also exposed to the ozone-default.xml, these properties were missing before.

## What is the link to the Apache JIRA

[HDDS-10691](https://issues.apache.org/jira/browse/HDDS-10691)
(Please replace this section with the link to the Apache JIRA)

## How was this patch tested?
green CI:
https://github.com/Galsza/ozone/actions/runs/8702230425

### CODE REVIEW

**Comment by Galsza** [comment]
@dombizita @fapifta please take a look

File: hadoop-hdds/common/src/main/resources/ozone-default.xml [hunk 1]
```diff
@@ -2238,17 +2238,35 @@
...
     </description>
   </property>
   <property>
     <name>hdds.key.dir.name</name>
     <value>keys</value>
     <tag>SCM, HDDS, X509, SECURITY</tag>
     <description>
-      Directory to store public/private key for SCM CA. This is relative to ozone/hdds meteadata dir.
+      Directory to store public/private key for SCM CA. This is relative to
+      ozone/hdds meteadata dir.
```
Comment (dombizita) [inline hadoop-hdds/common/src/main/resources/ozone-default.xml:2269]: Could you remove this? I don't this it is related to this change, you only broke the line into two lines.

File: hadoop-hdds/common/src/main/resources/ozone-default.xml [hunk 2]
```diff
@@ -2238,17 +2238,35 @@
...
+  </property>
+  <property>
+    <name>hdds.security.provider</name>
+    <value>BC</value>
+    <tag>OZONE, HDDS, X509, SECURITY, CRYPTO_COMPLIANCE</tag>
     <description>
-      SCM CA key length.  This is an algorithm-specific metric, such as modulus length, specified in number of bits.
+      The main security provider used for various cryptographic algorithms.
     </description>
   </property>
```
Comment (dombizita) [inline hadoop-hdds/common/src/main/resources/ozone-default.xml:2262]: Could you highlight in the PR description that you also expose already existing configs to the ozone-default.xml with this patch? I'll do some other already existing config addition to the ozone-default.xml (with the `CRYPTO_COMPLIANCE` tag) in [HDDS-10732](https://issues.apache.org/jira/browse/HDDS-10732).

**Review by dombizita — COMMENTED** [review]
Thanks for working on this @Galsza!

File: hadoop-hdds/common/src/main/resources/ozone-default.xml (hunk 2 above)
Comment (Galsza) [reply hadoop-hdds/common/src/main/resources/ozone-default.xml:2262]: Thank you @dombizita for the review, I've addressed your comments

**Review by fapifta — APPROVED** [review]
Thank you @Galsza for the proposed changes, LGTM.

**Pull request merged**